### PR TITLE
[FIXED JENKINS-29096] Advice users when they don't use the right hook url

### PR DESCRIPTION
**Pull request by fbelzunc**
The idea is to alert users with a logging when they don't use the right hook url in Bitbucket. Usually, they forget the last slash in `/bitbucket-hook/` and it seems this produces the stacktrace of the Jenkins issue.

https://issues.jenkins-ci.org/browse/JENKINS-29096

I was not able to re-produce the issue, but I think it should be fine as Stéphane Este-Gracias  recently faced the issue and told us the solution in the Jenkins issue.

@reviewbybees

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/BitbucketHookReceiver.java [hunk 1]
```diff
@@ -39,16 +41,21 @@ public String getUrlName() {
      */
     public void doIndex(StaplerRequest req) throws IOException {
         String body = IOUtils.toString(req.getInputStream());
```
Comment (tfennelly) [inline src/main/java/com/cloudbees/jenkins/plugins/BitbucketHookReceiver.java]: I know this is not part of this change, but I think this `toString` should be honouring the content char encoding.

**Comment by tfennelly** [comment]
:bee:

**Comment by amuniz** [comment]
:bee:

**Comment by fbelzunc** [comment]
@reviewbybees done

**Pull request merged**